backend/src/agents/esg_report_agent.py

```diff
@@ -0,0 +1,22 @@
+from src.agents import Agent, agent
+from src.prompts import PromptEngine
+
+engine = PromptEngine()
+
+
+@agent(
+    name="ESGReportAgent",
+    description="This agent is responsible for generating an ESG report",
+    tools=[],
+)
+class ESGReportAgent(Agent):
+    async def invoke(self, utterance: str) -> str:
+        user_prompt = engine.load_prompt(
+            "create-esg-report-user-prompt",
+            document_text=utterance)
```

Comment:
It feels a bit funny that we need to pass in the file via `utterance` and we need to annotate with `@agent`, our base Agent class is purpose built around the core "chat" style functionality of InferESG.

This is Something for another ticket entirely - but it feels like we need a new concept here, possibly creating a `SystemAgent` or `ReportAgent` base class that makes a cleaner separation from the `Agent` class - or possibly work the other way and change `Agent` class as it is now into `ChatAgent`.

What you've built here looks good and matches the designs we agreed on for this ticket

Reply:
I actually tried renaming the variable to something more sensible, but the linter didn't seem to like it.